lib/matplotlib/legend.py

```diff
@@ -507,10 +512,22 @@ def __init__(self, parent, handles, labels,
         self._loc_used_default = tmp  # ignore changes done by _set_loc
 
         # figure out title fontsize:
-        if title_fontsize is None:
-            title_fontsize = mpl.rcParams['legend.title_fontsize']
-        tprop = FontProperties(size=title_fontsize)
-        self.set_title(title, prop=tprop)
+        if title_prop is None:
+            if title_fontsize is None:
+                title_fontsize = mpl.rcParams["legend.title_fontsize"]
+            self.title_prop = FontProperties(
+                size=title_fontsize)
+        else:
+            self.title_prop = FontProperties._from_any(title_prop)
+            if isinstance(title_prop, dict) and "size" not in title_prop:
+                if title_fontsize is None:
+                    title_fontsize = mpl.rcParams["legend.title_fontsize"]
+                self.title_prop.set_size(title_fontsize)
```

Comment:
Thanks for the tests.  It would be good to make sure all the patch are tested a) so we understand the behaviour and b) so no one removes it later...

Reply:
I added one more test for Line 528, but can't get the test for Line 524 right. I checked `mpl.rcParams['legend.title_fontsize']` but find it is set to `None`. So the test `assert leg.get_title().get_fontsize() == mpl.rcParams['legend.title_fontsize']` will always fail.

Reply:
So now line 518 needs a check.  But I think this is very close!  Thanks for your patience...

Reply:
I just added one more test for L518. In this case when rcParams['legend.title_fontsize'] is `None`, the title fontsize will be defaulted to rcParams['font.size'] (when a FontProperties object is created with default args).